tests/options/sortAttributes/sort-attributes.test.ts

```diff
@@ -38,19 +39,181 @@ describe('Options', () => {
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
 			);
 
 			expect(actual).toStrictEqual(expected);
 		});
 		test('compare 2', () => {
+			const pugSortAttributes: SortAttributes = 'as-is';
+			const pugSortAttributesBeginning: string[] = [];
+			const pugSortAttributesEnd = ['v-for', ':key', 'src', 'alt', '@click', ':disabled'];
 			const expected: ReadonlyArray<string> = ['v-for', ':key', 'src', 'alt', '@click', ':disabled'];
 			const code: string[] = ['v-for', ':disabled', ':key', '@click', 'src', 'alt'];
-			const actual: string[] = code.sort((a, b) =>
-				compareAttributeToken(createAttributeToken(a), createAttributeToken(b), ['@click', ':disabled'], true)
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
+			);
+
+			expect(actual).toStrictEqual(expected);
+		});
+		test('compare 3', () => {
+			const pugSortAttributes: SortAttributes = 'as-is';
+			const pugSortAttributesBeginning: string[] = ['^x$', '^y$', '^z$'];
+			const pugSortAttributesEnd: string[] = ['v-for', ':key', 'src', 'alt', '@click', ':disabled'];
+			const expected: ReadonlyArray<string> = [
+				'x',
+				'y',
+				'z',
+				'c',
+				'a',
+				'b',
+				'v-for',
+				':key',
+				'src',
+				'alt',
+				'@click',
+				':disabled'
+			];
+			const code: string[] = ['y', 'c', 'z', 'a', ':disabled', 'alt', 'b', ':key', 'v-for', '@click', 'src', 'x'];
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
+			);
+
+			expect(actual).toStrictEqual(expected);
+		});
+		test('compare 4', () => {
+			const pugSortAttributes: SortAttributes = 'asc';
+			const pugSortAttributesBeginning: string[] = ['^x$', '^y$', '^z$'];
+			const pugSortAttributesEnd: string[] = ['v-for', ':key', 'src', 'alt', '@click', ':disabled'];
+			const expected: ReadonlyArray<string> = [
+				'x',
+				'y',
+				'z',
+				'D',
+				'a',
+				'b',
+				'c',
+				'v-for',
+				':key',
+				'src',
+				'alt',
+				'@click',
+				':disabled'
+			];
+			const code: string[] = [
+				'y',
+				'c',
+				'z',
+				'a',
+				':disabled',
+				'alt',
+				'b',
+				'D',
+				':key',
+				'v-for',
+				'@click',
+				'src',
+				'x'
+			];
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
+			);
+
+			expect(actual).toStrictEqual(expected);
+		});
+		test('compare 5', () => {
```

Comment:
I know it's hard, but I thing we should try to find names for all these `compare x` tests
Now there are 7 and we cannot read from them what they are doing ^^

Reply:
@SkyaTura Nice one, couldn't make it better ✨